lib/pbench/cli/server/user_delete.py

```diff
@@ -0,0 +1,65 @@
+import click
+from pbench import BadConfig
+from pbench.server import PbenchServerConfig
+from pbench.cli.server import pass_cli_context
+from pbench.cli.server.options import common_options
+from pbench.common.logger import get_pbench_logger
+from pbench.server.database.models.users import User
+from pbench.server.database.database import Database
+
+_NAME_ = "pbench-user-delete"
+
+
+class UserDelete:
+    def __init__(self, context):
+        self.context = context
+
+    def execute(self):
+        config = PbenchServerConfig(self.context.config)
+
+        logger = get_pbench_logger(_NAME_, config)
+
+        # We're going to need the Postgres DB to track dataset state, so setup
+        # DB access.
+        Database.init_db(config, logger)
+
+        # Determine if the token belongs to the admin user
+        user = User.query(username=self.context.username)
+
+        # Check if the user has the right privileges
+        if user is None or not user.is_admin():
+            raise RuntimeError(f"Not authorized to delete user {self.context.username}")
```

Comment:
Hmm. Why bother? More on that in subsequent comments. If we want to pretend to be doing authorization checks here, we could do it by adding a `--password` along with `--username` and matching the password against the DB hash; but since anyone operating on the server CLI can fire up `python3`, import a few of our modules, and access the DB directly this seems like just an annoying speedbump rather than any real security.